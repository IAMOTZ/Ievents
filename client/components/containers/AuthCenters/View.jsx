import React from 'react';
import PropTypes from 'prop-types';
import SideNavigation from '../../common/SideNavigation';
import Header from '../../common/Header';
import CenterCards from '../../common/CenterCards';
import { LoadingBox } from '../../common/LoadingAnimation';
import { AuthTopNavigation } from '../../common/TopNavigation';

const View = props => (
  <div id="auth-centers-container">
    <AuthTopNavigation
      name={props.userName}
      title="Centers"
      isAdmin={props.isAdmin}
      isSuperAdmin={props.isSuperAdmin}
      dispatch={props.dispatch}
    />
    <div className="container-fluid">
      <div className="row">
        <SideNavigation
          name={props.userName}
          isAdmin={props.isAdmin}
          isSuperAdmin={props.isSuperAdmin}
          dispatch={props.dispatch}
        />
        <div className="col-lg-10 offset-lg-2 mt-lg-0">
          <Header text="Centers" />
          <div className="page-content">
            {
              props.centers.length === 0 && props.fetchingCentersStarted ?
                <LoadingBox iconSize={4} /> :
                <div className="container">
                  <div className="mt-5">
                    <div className="row">
                      <CenterCards
                        centers={props.centers}
                        isAdmin={props.isAdmin || props.isSuperAdmin}
                        onBook={props.onBook}
                        onEdit={props.onEdit}
                      />
                    </div>
                  </div>
                </div>
            }
          </div>
        </div>
      </div>
    </div>
  </div>
);

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  userName: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  centers: PropTypes.array.isRequired,
  fetchingCentersStarted: PropTypes.bool.isRequired,
  onEdit: PropTypes.func.isRequired,
  onBook: PropTypes.func.isRequired,
};

export default View;
