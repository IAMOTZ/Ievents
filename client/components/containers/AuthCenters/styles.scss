#auth-centers-container {
  .page-content {
    min-height: 100vh;
    margin-bottom: 50px;
  }
}