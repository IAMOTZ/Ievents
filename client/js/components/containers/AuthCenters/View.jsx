import React from 'react';
import PropTypes from 'prop-types';
import SideNavigation from '../../common/SideNavigation';
import Header from '../../common/Header';
import CenterCards from '../../common/CenterCards';
import { CenterModal } from '../../common/Modals';
import { LoadingBox } from '../../common/LoadingAnimation';
import { AuthTopNavigation } from '../../common/TopNavigation';

const View = props => (
  <div id="centers2-container">
    <AuthTopNavigation
      name={props.userName}
      title="Centers"
      isAdmin={props.isAdmin}
      isSuperAdmin={props.isSuperAdmin}
      dispatch={props.dispatch}
    />
    <div className="container-fluid">
      <div className="row">
        <SideNavigation
          name={props.userName}
          isAdmin={props.isAdmin}
          isSuperAdmin={props.isSuperAdmin}
          dispatch={props.dispatch}
        />
        <div className="col-lg-10 offset-lg-2 mt-lg-0" id="main-content">
          <Header text="Centers" />
          {
            props.centers.length === 0 && props.fetchingCentersStarted ?
              <LoadingBox iconSize={4} /> :
              <section id="centers-section">
                <div className="container">
                  {/* Search Box */}
                  <div
                    id="center-section-content"
                    className="d-flex flex-column align-items-center"
                  >
                    <div className="input-group w-50 mt-lg-5 search-box">
                      <input
                        type="text"
                        className="form-control"
                        placeholder="Search for a center"
                      />
                      <div className="input-group-btn d-none d-sm-block">
                        <button
                          type="button"
                          className="btn btn-secondary dropdown-toggle"
                          data-toggle="dropdown"
                        >Search with
                        </button>
                        <div className="dropdown-menu dropdown-menu-right">
                          <a className="dropdown-item" href="#">Location</a>
                          <div role="separator" className="dropdown-divider" />
                          <a className="dropdown-item" href="#">Name</a>
                        </div>
                      </div>
                    </div>
                    <div className="search-box mt-3 d-block d-sm-none">
                      <button
                        type="button"
                        className="btn btn-secondary dropdown-toggle"
                        data-toggle="dropdown"
                      >Search with
                      </button>
                      <div className="dropdown-menu">
                        <a className="dropdown-item" href="#">Location</a>
                        <div role="separator" className="dropdown-divider" />
                        <a className="dropdown-item" href="#">Name</a>
                      </div>
                    </div>
                  </div>
                  <div className="mt-5">
                    <div className="row">
                      <CenterCards
                        centers={props.centers}
                        btnAction={props.showModal}
                        editAction={props.onEdit}
                        isAdmin={props.isAdmin || props.isSuperAdmin}
                      />
                    </div>
                  </div>
                </div>
              </section>
          }
          {/* Modal */}
          {
            props.modalContent ? <CenterModal
              modalContent={props.modalContent}
              onBook={props.onBook}
              redirectPath="/users/login"
            /> : null
          }
        </div>
      </div>
    </div>
  </div>
);

View.defaultProps = {
  modalContent: {},
};

/* eslint-disable react/forbid-prop-types */
View.propTypes = {
  userName: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isSuperAdmin: PropTypes.bool.isRequired,
  dispatch: PropTypes.func.isRequired,
  centers: PropTypes.array.isRequired,
  fetchingCentersStarted: PropTypes.bool.isRequired,
  showModal: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  modalContent: PropTypes.object,
  onBook: PropTypes.func.isRequired,
};

export default View;
