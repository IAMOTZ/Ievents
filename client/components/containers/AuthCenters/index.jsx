import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  getAllCenters, setCenterToUpdate,
} from '../../../actions/centerActions';
import { setCenterToBook } from '../../../actions/eventActions';
import './styles.scss';
import View from './View';

@connect((store) => {
  const { user } = store.authReducer;
  return {
    userName: user.name,
    isAdmin: user.role === 'admin' || user.role === 'superAdmin',
    isSuperAdmin: user.role === 'superAdmin',
    fetchingCentersStarted: store.fetchCentersReducer.fetchingCenterStarted,
    centers: store.fetchCentersReducer.centers,
  };
})
class AuthCenters extends React.Component {
  componentWillMount() {
    this.props.dispatch(getAllCenters());
  }

  /**
   * It updates the store about a center that is to be edited.
   * @param {Event} event The event object.
   */
  onEdit = (event) => {
    this.props.dispatch(setCenterToUpdate(Number(event.target.id)));
  }

  /**
   * It updates the store about a center that is to be booked.
   * @param {Event} event The event object.
   */
  onBook = (event) => {
    this.props.dispatch(setCenterToBook(Number(event.target.id)));
  }

  render() {
    return (
      <View
        userName={this.props.userName}
        isAdmin={this.props.isAdmin}
        isSuperAdmin={this.props.isSuperAdmin}
        dispatch={this.props.dispatch}
        centers={this.props.centers}
        fetchingCentersStarted={this.props.fetchingCentersStarted}
        onEdit={this.onEdit}
        onBook={this.onBook}
      />
    );
  }
}

AuthCenters.defaultProps = {
  userName: '',
  isAdmin: false,
  isSuperAdmin: false,
  fetchingCentersStarted: false,
  centers: [],
  dispatch: () => {},
};

/* eslint-disable react/forbid-prop-types */
AuthCenters.propTypes = {
  userName: PropTypes.string,
  isAdmin: PropTypes.bool,
  isSuperAdmin: PropTypes.bool,
  fetchingCentersStarted: PropTypes.bool,
  centers: PropTypes.array,
  dispatch: PropTypes.func,
};

export default AuthCenters;
