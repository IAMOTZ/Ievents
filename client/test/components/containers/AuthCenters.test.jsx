import React from 'react';
import AuthCenters from '../../../components/containers/AuthCenters';
import AuthCentersView from '../../../components/containers/AuthCenters/View';

const sampleCenter = {
  id: 1,
  name: 'test-center1',
  location: 'test-center-location',
  details: 'test-center-details',
  capacity: '200',
  price: '500',
  images: ['http://testImage.jpg'],
};

describe('<RegularCenters />', () => {
  const props = {
    isAdmin: false,
    isSuperAdmin: false,
    fetchingCenterStarted: false,
    centers: [sampleCenter],
    pagination: {},
    dispatch: jest.fn(),
  };
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AuthCenters {...props} />);
    });
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Behaviour:', () => {
    let wrapper; let wrapperInstance;
    beforeEach(() => {
      props.dispatch.mockReset();
      wrapper = shallow(<AuthCenters {...props} />);
      wrapperInstance = wrapper.instance();
    });
    // The names on the following test block refers
    // to the instance method that handles the behaviour tested.
    describe('COMOPONENT_WILL_MOUNT', () => {
      it('should dispatch an action', () => {
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });

    describe('ON_EDIT', () => {
      it('should dispatch an action', () => {
        const fakeEvent = { target: { id: sampleCenter.id } };
        props.dispatch.mockReset();
        wrapperInstance.onEdit(fakeEvent);
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });

    describe('ON_BOOK', () => {
      it('should dispatch an action', () => {
        const fakeEvent = { target: { id: sampleCenter.id } };
        props.dispatch.mockReset();
        wrapperInstance.onBook(fakeEvent);
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });

    describe('ON_VIEW_TRANSACTIONS', () => {
      it('should dispatch an action', () => {
        const fakeEvent = { currentTarget: { id: sampleCenter.id } };
        props.dispatch.mockReset();
        wrapperInstance.onViewTransactions(fakeEvent);
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });

    describe('CREATE_MODAL_CONTENT', () => {
      it('should update the state with the content of a modal', () => {
        wrapperInstance.createModalContent(sampleCenter);
        expect(wrapper.state('modalContent')).toEqual({
          centerName: sampleCenter.name,
          centerImages: sampleCenter.images,
          centerDetails: sampleCenter.details
        });
      });
    });

    describe('UPDATE_PAGINATION', () => {
      it('should dispatch an action', () => {
        const fakePageData = { selected: 3 };
        props.dispatch.mockReset();
        wrapperInstance.updatePagination(fakePageData);
        expect(props.dispatch.mock.calls.length).toEqual(1);
      });
    });
  });
});

describe('<AuthCentersView />', () => {
  const props = {
    userName: 'tester',
    isAdmin: false,
    isSuperAdmin: false,
    dispatch: jest.fn(),
    isTransactionsPage: false,
    centers: [],
    fetchingCenterStarted: false,
    onEdit: jest.fn(),
    onBook: jest.fn(),
    onViewTransactions: jest.fn(),
    modalContent: {},
    createModalContent: jest.fn(),
    pagination: {},
    updatePagination: jest.fn(),
  };
  const alterProps = newProps => ({ ...props, ...newProps });
  describe('Rendering', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AuthCentersView {...props} />);
    });
    it('should render correctly when there are no centers and it is fetching', () => {
      wrapper.setProps(alterProps({ fetchingCentersStarted: true }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly when there are centers and it is not fetching', () => {
      wrapper.setProps(alterProps({ centers: [sampleCenter] }));
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly on transactions page', () => {
      wrapper.setProps(alterProps({ centers: [sampleCenter], isTransactionsPage: true }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
