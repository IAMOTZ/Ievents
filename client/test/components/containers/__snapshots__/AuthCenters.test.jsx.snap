// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthCentersView /> Rendering should render correctly on transactions page 1`] = `
<div
  id="auth-centers-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="Transactions"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-lg-2 mt-lg-0"
      >
        <Header
          text="Transactions"
        />
        <div
          className="page-content"
        >
          <div
            className="container"
          >
            <div>
              <div
                className="row"
              >
                <CenterCards
                  centers={
                    Array [
                      Object {
                        "capacity": "200",
                        "details": "test-center-details",
                        "id": 1,
                        "images": Array [
                          "http://testImage.jpg",
                        ],
                        "location": "test-center-location",
                        "name": "test-center1",
                        "price": "500",
                      },
                    ]
                  }
                  createModalContent={[MockFunction]}
                  isAdmin={false}
                  isTransactionsPage={true}
                  onBook={[MockFunction]}
                  onEdit={[MockFunction]}
                  onViewTransactions={[MockFunction]}
                />
              </div>
            </div>
          </div>
          <Pagination
            onPageChange={[MockFunction]}
            pageCount={NaN}
          />
          <CenterDetailsModal
            centerDetails=""
            centerImages={
              Array [
                "/images/defaultImgx4.jpeg",
              ]
            }
            centerName=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AuthCentersView /> Rendering should render correctly when there are centers and it is not fetching 1`] = `
<div
  id="auth-centers-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="Centers"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-lg-2 mt-lg-0"
      >
        <Header
          text="Centers"
        />
        <div
          className="page-content"
        >
          <div
            className="container"
          >
            <h1
              className="caption-text"
            >
              A special center for a special event
            </h1>
            <div>
              <div
                className="row"
              >
                <CenterCards
                  centers={
                    Array [
                      Object {
                        "capacity": "200",
                        "details": "test-center-details",
                        "id": 1,
                        "images": Array [
                          "http://testImage.jpg",
                        ],
                        "location": "test-center-location",
                        "name": "test-center1",
                        "price": "500",
                      },
                    ]
                  }
                  createModalContent={[MockFunction]}
                  isAdmin={false}
                  isTransactionsPage={false}
                  onBook={[MockFunction]}
                  onEdit={[MockFunction]}
                  onViewTransactions={[MockFunction]}
                />
              </div>
            </div>
          </div>
          <Pagination
            onPageChange={[MockFunction]}
            pageCount={NaN}
          />
          <CenterDetailsModal
            centerDetails=""
            centerImages={
              Array [
                "/images/defaultImgx4.jpeg",
              ]
            }
            centerName=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AuthCentersView /> Rendering should render correctly when there are no centers and it is fetching 1`] = `
<div
  id="auth-centers-container"
>
  <AuthTopNavigation
    dispatch={[MockFunction]}
    isAdmin={false}
    isSuperAdmin={false}
    name="tester"
    title="Centers"
  />
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <SideNavigation
        dispatch={[MockFunction]}
        isAdmin={false}
        isSuperAdmin={false}
        name="tester"
      />
      <div
        className="col-lg-10 offset-lg-2 mt-lg-0"
      >
        <Header
          text="Centers"
        />
        <div
          className="page-content text-center"
        >
          <h1
            className="display-3"
          >
            The centers list is empty
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<RegularCenters /> Rendering: should render correctly 1`] = `
<View
  centers={
    Array [
      Object {
        "capacity": "200",
        "details": "test-center-details",
        "id": 1,
        "images": Array [
          "http://testImage.jpg",
        ],
        "location": "test-center-location",
        "name": "test-center1",
        "price": "500",
      },
    ]
  }
  createModalContent={[Function]}
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          [Function],
        ],
      ],
    }
  }
  fetchingCenterStarted={false}
  isAdmin={false}
  isSuperAdmin={false}
  isTransactionsPage={false}
  modalContent={null}
  onBook={[Function]}
  onEdit={[Function]}
  onViewTransactions={[Function]}
  pagination={Object {}}
  updatePagination={[Function]}
  userName=""
/>
`;
